apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c

```diff
@@ -722,13 +800,65 @@ static void utc_audio_pcm_writei_p(void)
 		}
 	} while (num_read > 0);
 
-	sleep(2);
-	printf("Playback done! Total Frames: %d\n", pcm_bytes_to_frames(g_pcm, total_frames));
+	if (buffer != NULL) {
+		free(buffer);
+		buffer = NULL;
+	}
+	if (fd > 0) {
+		close(fd);
+	}
 
-	clean_all_data(fd, buffer);
+}
+
+static void utc_audio_pcm_drain_p(void)
+{
+	/* Executed after writei positive tc */
+
+	int ret;
+	ret = pcm_drain(g_pcm);
```

Comment:
I think we'd better put code as we give notice to users in readi test with gets

Reply:
I think giving notice with gets will defeat the purpose of pcm_drain. Because if the user does not give input for a long time, then all the buffers will be played and pcm_drain will not be verified since there is no pending buffer.